frontend-react/src/layouts/MainLayout.tsx

```diff
@@ -0,0 +1,43 @@
+import { GovBanner } from "@trussworks/react-uswds";
+import classNames from "classnames";
+import { Outlet, useMatches } from "react-router-dom";
+import { ToastContainer } from "react-toastify";
+
+import App from "../App";
+import { DAPHeader } from "../components/header/DAPHeader";
+import { ReportStreamHeader } from "../components/header/ReportStreamHeader";
+import { ReportStreamFooter } from "../components/ReportStreamFooter";
+import SenderModeBanner from "../components/SenderModeBanner";
+import { USLink } from "../components/USLink";
+import { useSessionContext } from "../contexts/SessionContext";
+
+const MainLayout = () => {
+    const { environment } = useSessionContext();
+    const matches = useMatches();
+    const { handle = {} } = matches.at(-1) ?? {};
```

Comment:
Will it _always_ be the last item? Do want to make it more agnostic with something like: `matches.find(match => match.handle)`?

Reply:
It is not well documented, but my assumption is that matches is the flat route tree path to the current route (the last match). Their breadcrumb example reinforces this.

Reply:
When I messed with your branch locally, `matches.find(match => match.handle)` worked fine as well, but if you're confident that `.at(-1)` will always give us the same result as well, all good.

Reply:
Your find algorithm would just return the first route that had handle defined, which is an invalid solution in this case. We want the route object for the current route we're rendering, which is always the last entry in the matches array.

Reply:
> first route that had handle defined

from my testing locally, there was only one route that had `handle` defined, and that's if is `isContentPage` is added to the route. if you want a more specific `find`, you could do: `matches.find(match => match.handle?.isContentPage`

Reply:
That is still not what we're trying to find. We want the current route being rendered. Whether handle or anything there-in exists for any ancestor route is irrelevant, because those aren't the current route being rendered.